declarative-examples/simple-examples/whenEnvTrue.groovy

```diff
@@ -0,0 +1,25 @@
+pipeline {
+  agent any
+
+  environment {
+      FOO = "BAR"
+  }
+
+  stages {
+    stage("Hello") {
+      steps {
+        echo "Hello"
+      }
+    }
+    stage("Evaluate FOO") {
+      when {
+        // stage won't be skipped as long as FOO == BAR
+        environment name: "FOO", value: "BAR"
```

Comment:
These examples would be better if the environment variable value were set dynamically and *sometimes* matched. "Never run this block" seems to be an unusual case.